openedx/core/djangoapps/user_api/accounts/views.py

```diff
@@ -89,14 +87,34 @@ def get(self, request, username):
         GET /api/user/v0/accounts/{username}/
         """
         try:
-            existing_user, existing_user_profile = self._get_user_and_profile(username)
+            account_settings = AccountView.get_serialized_account(username)
         except AccountUserNotFound:
             return Response(status=status.HTTP_404_NOT_FOUND)
 
+        return Response(account_settings)
+
+    @staticmethod
+    def get_serialized_account(username):
+        """Returns the user's account information serialized as JSON.
+
+        Note:
+          This method does not perform authentication so it is up to the caller
+          to ensure that the result is only returned to the appropriate user.
```

Comment:
It seems like overkill to me to have to document all the rules here. The intention is that the caller has done the authorization and determined that is is appropriate to return the account. There are also callers that don't expose the result so don't have to perform any authorization. I'd rather have the authorization scheme documented in the code that does it.